File: android/app/src/main/java/com/magicforms/mobile/data/models/AuthModels.kt
package com.magicforms.mobile.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ApiUser(
    val id: Int,
    val username: String,
    val email: String = "",
    @SerialName("first_name") val firstName: String = "",
    @SerialName("last_name") val lastName: String = "",
    @SerialName("full_name") val fullName: String = "",
    @SerialName("is_staff") val isStaff: Boolean = false,
    @SerialName("is_superuser") val isSuperuser: Boolean = false,
) {
    val displayName: String
        get() = when {
            fullName.isNotBlank() -> fullName
            firstName.isNotBlank() || lastName.isNotBlank() ->
                listOf(firstName, lastName).filter { it.isNotBlank() }.joinToString(" ")
            else -> username
        }
}

@Serializable
data class ApiEntity(
    val id: Int? = null,
    val slug: String,
    val name: String,
    @SerialName("logo_url") val logoUrl: String? = null,
)

@Serializable
data class LoginRequest(
    val username: String,
    val password: String,
    @SerialName("entity_slug") val entitySlug: String? = null,
)

@Serializable
data class LoginResponse(
    val ok: Boolean,
    val token: String? = null,
    @SerialName("expires_at") val expiresAt: String? = null,
    @SerialName("token_type") val tokenType: String? = null,
    @SerialName("login_via") val loginVia: String? = null,
    val user: ApiUser? = null,
    val entities: List<ApiEntity>? = null,
)

@Serializable
data class MeResponse(
    val ok: Boolean,
    val user: ApiUser? = null,
    val entities: List<ApiEntity>? = null,
)

@Serializable
data class DirectoryEntitiesResponse(
    val ok: Boolean,
    val entities: List<ApiEntity>? = null,
)

@Serializable
data class ApiErrorBody(
    val ok: Boolean? = null,
    val error: String? = null,
    val message: String? = null,
    val entities: List<ApiEntity>? = null,
)

class ApiException(
    val code: String,
    override val message: String,
    val entities: List<ApiEntity>? = null,
) : Exception(message)

File: ios/MagicFormsMobile/Models/AuthModels.swift
import Foundation

struct APIUser: Codable, Identifiable, Equatable {
    let id: Int
    let username: String
    let email: String
    let firstName: String
    let lastName: String
    let fullName: String
    let isStaff: Bool
    let isSuperuser: Bool

    enum CodingKeys: String, CodingKey {
        case id, username, email
        case firstName = "first_name"
        case lastName = "last_name"
        case fullName = "full_name"
        case isStaff = "is_staff"
        case isSuperuser = "is_superuser"
    }

    var displayName: String {
        if !fullName.isEmpty { return fullName }
        if !firstName.isEmpty || !lastName.isEmpty {
            return [firstName, lastName].filter { !$0.isEmpty }.joined(separator: " ")
        }
        return username
    }
}

struct APIEntity: Codable, Identifiable, Equatable, Hashable {
    let id: Int?
    let slug: String
    let name: String
    let logoUrl: String? = nil

    enum CodingKeys: String, CodingKey {
        case id, slug, name
        case logoUrl = "logo_url"
    }
}

struct LoginRequest: Encodable {
    let username: String
    let password: String
    let entitySlug: String?

    enum CodingKeys: String, CodingKey {
        case username, password
        case entitySlug = "entity_slug"
    }
}

struct LoginResponse: Decodable {
    let ok: Bool
    let token: String?
    let expiresAt: String?
    let tokenType: String?
    let loginVia: String?
    let user: APIUser?
    let entities: [APIEntity]?

    enum CodingKeys: String, CodingKey {
        case ok, token, user, entities
        case expiresAt = "expires_at"
        case tokenType = "token_type"
        case loginVia = "login_via"
    }
}

struct MeResponse: Decodable {
    let ok: Bool
    let user: APIUser?
    let entities: [APIEntity]?
}

struct DirectoryEntitiesResponse: Decodable {
    let ok: Bool
    let entities: [APIEntity]?
}

struct APIErrorBody: Decodable {
    let ok: Bool?
    let error: String?
    let message: String?
    let entities: [APIEntity]?
}

struct APIError: LocalizedError {
    let code: String
    let message: String
    let entities: [APIEntity]?

    var errorDescription: String? { message }
}
